PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidNativeAd.java

```diff
@@ -0,0 +1,346 @@
+import java.util.List;
+
+public class PrebidNativeAd {
+    private String title;
+    private String description;
+    private String iconUrl;
+    private String imageUrl;
+    private String cta;
+    private String sponsoredBy;
+    private String clickUrl;
+    private ArrayList<String> imp_trackers;
+    private VisibilityDetector visibilityDetector;
+    private boolean expired;
+    private View registeredView;
+    private PrebidNativeAdEventListener listener;
+    private ArrayList<ImpressionTracker> impressionTrackers;
+
+
+    static PrebidNativeAd create(String cacheId) {
+        String content = CacheManager.get(cacheId);
+        if (!TextUtils.isEmpty(content)) {
+            try {
+                JSONObject details = new JSONObject(content);
+                String admStr = details.getString("adm");
+                JSONObject adm = new JSONObject(admStr);
+                JSONArray asset = adm.getJSONArray("assets");
+                final PrebidNativeAd ad = new PrebidNativeAd();
+                CacheManager.registerCacheExpiryListener(cacheId, new CacheManager.CacheExpiryListener() {
+                    @Override
+                    public void onCacheExpired() {
+                        if (ad.registeredView == null) {
+                            if (ad.listener != null) {
+                                ad.listener.onAdExpired();
+                            }
+                            ad.expired = true;
+                            if (ad.visibilityDetector != null) {
+                                ad.visibilityDetector.destroy();
+                                ad.visibilityDetector = null;
+                            }
+                            ad.impressionTrackers = null;
+                            ad.listener = null;
+                        }
+                    }
+                });
+                for (int i = 0; i < asset.length(); i++) {
+                    JSONObject adObject = asset.getJSONObject(i);
+                    if (adObject.has("title")) {
+                        JSONObject title = adObject.getJSONObject("title");
+                        if (title.has("text")) {
+                            ad.setTitle(title.getString("text"));
+                        }
+                    }
+                    if (adObject.has("data")) {
+                        JSONObject data = adObject.getJSONObject("data");
+                        if (data.optInt("type") == 1) {
+                            ad.setSponsoredBy(data.getString("value"));
+                        } else if (data.optInt("type") == 2) {
+                            ad.setDescription(data.getString("value"));
+                        } else if (data.optInt("type") == 12) {
+                            ad.setCallToAction(data.getString("value"));
+                        }
+                    }
+
+                    if (adObject.has("img")) {
+                        if (adObject.optInt("id") == 1) {
```

Comment:
hi @avohraa 

Could you clarify, please, why did you use the "id" field to determine the object type?